src/io_handler.c

```diff
@@ -41,13 +49,30 @@ int extractsf_handle_args(int argc, char ** argv, bool * use_fmt, int * nevents,
     return check_root_filename(* input_file);
 }
 
-int hipo2root_handle_args(int argc, char ** argv, char ** input_file, int * run_no) {
+int hipo2root_handle_args(int argc, char ** argv, char ** input_file, int * run_no, bool * use_simul) {
+    // Handle optional arguments.
+    int opt;
+    while ((opt = getopt(argc, argv, "s")) != -1) {
+        switch (opt) {
+            case 's': * use_simul = true;         break;
+            default:  return 1; // Bad usage of optional arguments.
+        }
+    }
+
+    // Handle positional arguments
     if (argc < 2) return 1;
     if (argc > 3) return 2;
 
-    * input_file = (char *) malloc(strlen(argv[1]) + 1);
-    strcpy(* input_file, argv[1]);
-    return handle_hipo_filename(* input_file, run_no);
+    * input_file = (char *) malloc(strlen(argv[argc - 1]) + 1);
```

Comment:
This is scary. What if I call the program using: `hipo2root FILENAME -s`? Wouldn't that break the program?

Edit. It only made sense before because the filename is the only argument the program receives.